integration/aggregate_documents_compat_test.go

```diff
@@ -1626,14 +1626,21 @@ func TestAggregateCompatProject(t *testing.T) {
 				bson.D{{"$sort", bson.D{{"_id", -1}}}},
 				bson.D{{"$project", bson.D{{"type", bson.D{{"$type", bson.D{{"$non-existent", "$v"}}}}}}}},
 			},
-			skip: "https://github.com/FerretDB/FerretDB/issues/2678",
+			resultType: emptyResult,
 		},
 		"TypeRecursiveInvalid": {
 			pipeline: bson.A{
 				bson.D{{"$sort", bson.D{{"_id", -1}}}},
 				bson.D{{"$project", bson.D{{"type", bson.D{{"$type", bson.D{{"v", "$v"}}}}}}}},
 			},
 		},
+		"TypeRecursiveArrayInvalid": {
```

Comment:
This test is the same as https://github.com/FerretDB/FerretDB/pull/2829/files#diff-8e3642cbc845cda87e6c35924b1170d8932453298b2c4552d662d6a9a04037a4R303

Any reason why we want to duplicate it in compat and integration test? :thinking:

Reply:
This test is different now. It should be recursive